sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/core/resource/AzureStorageBlobProtocolResolver.java

```diff
@@ -4,56 +4,102 @@
 package com.azure.spring.core.resource;
 
 import com.azure.storage.blob.BlobServiceClient;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.springframework.beans.BeansException;
-import org.springframework.beans.factory.config.BeanFactoryPostProcessor;
-import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
-import org.springframework.context.ResourceLoaderAware;
-import org.springframework.core.io.DefaultResourceLoader;
+import com.azure.storage.blob.models.BlobContainerItem;
+import com.azure.storage.blob.models.BlobContainerListDetails;
+import com.azure.storage.blob.models.BlobListDetails;
+import com.azure.storage.blob.models.ListBlobContainersOptions;
+import com.azure.storage.blob.models.ListBlobsOptions;
+import java.util.stream.Stream;
 import org.springframework.core.io.ProtocolResolver;
 import org.springframework.core.io.Resource;
-import org.springframework.core.io.ResourceLoader;
 
 /**
  * A {@link ProtocolResolver} implementation for the {@code azure-blob://} protocol.
- *
  */
-public class AzureStorageBlobProtocolResolver implements ProtocolResolver, ResourceLoaderAware,
-                                                         BeanFactoryPostProcessor {
+public class AzureStorageBlobProtocolResolver extends AbstractAzureStorageProtocolResolver {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(AzureStorageBlobProtocolResolver.class);
-
-    private ConfigurableListableBeanFactory beanFactory;
     private BlobServiceClient blobServiceClient;
 
+    private static final BlobListDetails RETRIEVE_NOTHING_DETAILS = new BlobListDetails();
+    private static final BlobContainerListDetails RETRIEVE_NOTHING_CONTAINER_DETAILS = new BlobContainerListDetails();
+
+    static {
+        RETRIEVE_NOTHING_DETAILS.setRetrieveCopy(false);
+        RETRIEVE_NOTHING_DETAILS.setRetrieveDeletedBlobs(false);
+        RETRIEVE_NOTHING_DETAILS.setRetrieveDeletedBlobsWithVersions(false);
+        RETRIEVE_NOTHING_DETAILS.setRetrieveImmutabilityPolicy(false);
+        RETRIEVE_NOTHING_DETAILS.setRetrieveMetadata(false);
+        RETRIEVE_NOTHING_DETAILS.setRetrieveLegalHold(false);
+        RETRIEVE_NOTHING_DETAILS.setRetrieveSnapshots(false);
+        RETRIEVE_NOTHING_DETAILS.setRetrieveTags(false);
+        RETRIEVE_NOTHING_DETAILS.setRetrieveUncommittedBlobs(false);
+        RETRIEVE_NOTHING_CONTAINER_DETAILS.setRetrieveMetadata(false);
+        RETRIEVE_NOTHING_CONTAINER_DETAILS.setRetrieveDeleted(false);
+    }
+
+    public AzureStorageBlobProtocolResolver(BlobServiceClient blobServiceClient) {
```

Comment:
Do we still need this constructor here?

Reply:
Actually, for now, we don't need this constructor.
Because our test cases are based on this constructor,  so I kept the constructor,
I will remove it and refactor the test cases.